code/game/machinery/constructable_frame.dm

```diff
@@ -670,6 +670,42 @@ to destroy them and players will be able to make replacements.
 	req_components = list(
 							/obj/item/stock_parts/manipulator = 2,
 							/obj/item/stock_parts/matter_bin = 1)
+	var/static/list/regular_components = list(
+							/obj/item/stock_parts/manipulator = 2,
+							/obj/item/stock_parts/matter_bin = 1)
+	//keep these the same as the corresponding machine names.
+	var/static/list/grinder_names_paths = list(
+							"All-In-One Grinder" = /obj/machinery/reagentgrinder/empty,
+							"Botanitank" = /obj/machinery/reagentgrinder/Botanitank/empty)
+
+/obj/item/circuitboard/reagentgrinder/screwdriver_act(mob/living/user, obj/item/I)
+	. = TRUE
+	if(!I.use_tool(src, user, FALSE, volume = I.tool_volume))
+		return
+	var/choice = tgui_input_list(user, "Circuit Setting", "What would you change the board setting to?", grinder_names_paths)
+	if(!choice)
+		return
+	set_type(user, choice)
+
+/obj/item/circuitboard/reagentgrinder/proc/set_type(mob/user, type)
+	if(!ispath(type))
+		board_name = type
+		type = grinder_names_paths[type]
+	else
+		for(var/name in grinder_names_paths)
+			if(grinder_names_paths[name] == type)
+				board_name = name
+
+
+				break
+	if(board_name == "Botanitank")
```

Comment:
Empty spaces,
698-700 delete
700-701 gap

Reply:
> Edit: I think? You can use a switch instead of a for loop?

Wouldn't that be a problem if the grinder ever gets more children? Also what would be the advantage?

Reply:
I mean, you can get rid of the for loop, and just use a switch, instead of doing 
`
for(var/name in ...)
  if(board_name == type)
    board_name = name

if(board_name == "botanik"
`

You can do
`switch(type)
    if("Botanik")
      ...
    if("all_in_one")
      ...
    board_name = name // if you need board_name
`

Advantage is optimization, and not sure if there is a need for for loop at all

Reply:
I suppose if there are ever enough types of grinder that a loop is necessary it can be added then. very well.